### PR TITLE
Clean up H3Index Reader/Creator

### PR DESCRIPTION
**Pull request by yupeng9**
Cleaned up H3Index Reader/Creator, added some docs. Also, changed the index config to include resolutions.

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/geospatial/H3IndexCreator.java [hunk 1]
```diff
@@ -303,7 +339,6 @@ public static void main(String[] args)
     Point point2 = GeometryUtils.GEOMETRY_FACTORY.createPoint(new Coordinate(37.368832, -122.036346));
```
Comment (kishoreg) [inline pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/geospatial/H3IndexCreator.java:339]: move the Main into a test case

File: pinot-core/src/main/java/org/apache/pinot/core/segment/index/datasource/BaseDataSource.java [hunk 2]
```diff
@@ -44,13 +44,10 @@
...
 
   public BaseDataSource(DataSourceMetadata dataSourceMetadata, ForwardIndexReader<?> forwardIndex,
-                        @Nullable Dictionary dictionary, @Nullable InvertedIndexReader<?> invertedIndex,
-                        @Nullable InvertedIndexReader<?> rangeIndex,
-                        @Nullable H3IndexReader h3Index,
-                        @Nullable TextIndexReader textIndex,
-                        @Nullable TextIndexReader fstIndex,
-                        @Nullable BloomFilterReader bloomFilter,
-                        @Nullable NullValueVectorReader nullValueVector) {
+      @Nullable Dictionary dictionary, @Nullable InvertedIndexReader<?> invertedIndex,
```
Comment (kishoreg) [inline pinot-core/src/main/java/org/apache/pinot/core/segment/index/datasource/BaseDataSource.java:47]: make sure you are using the stylesheet to format the code

File: pinot-tools/src/main/resources/examples/batch/starbucksStores/starbucksStores_offline_table_config.json [hunk 3]
```diff
@@ -15,8 +15,11 @@
     "loadMode": "MMAP",
     "invertedIndexColumns": [
     ],
-    "h3IndexColumns": [
-      "location_st_point"
+    "h3IndexConfigs": [
```
Comment (kishoreg) [inline pinot-tools/src/main/resources/examples/batch/starbucksStores/starbucksStores_offline_table_config.json:18]: I think we added a new way of adding indexing configs for each field... can you follow the pattern used by text index? can be done in another PR

**Review by kishoreg — APPROVED** [review]

File: pinot-core/src/main/java/org/apache/pinot/core/segment/index/datasource/BaseDataSource.java (hunk 2 above)
Comment (yupeng9) [reply pinot-core/src/main/java/org/apache/pinot/core/segment/index/datasource/BaseDataSource.java:47]: Yes, this is done via auto-formatting.

**Pull request merged**